### PR TITLE
do not generate device token if 2FA is enable for user

### LINKED ISSUE #24976: Clients must not be able to generate a device token if 2FA is enabled
### Steps to reproduce
1. Enable a 2FA provider
2. Send a request with valid username and password to [/token/generate](https://github.com/owncloud/core/blob/2d5bee0661d52629b1ca0b8ec3c57752a3b7d8ed/core/routes.php#L50)

### Expected behaviour
Something like 400, 401 and __no__ new token.

### Actual behaviour
A new token is generated which can then be used to access user data (token auth is always allowed) :boom: 

**ownCloud version:**
9.1.0

### PR DESCRIPTION
**Pull request by ChristophWurst**
fixes #24976

Can be tested with `curl -v --request POST -d user=admin -d password=admin localhost:8080/index.php/token/generate`

@PVince81 @rullzer @nickvergessen review please

### CODE REVIEW

**Comment by PVince81** [comment]
:+1:

File: core/Controller/TokenController.php [hunk 1]
```diff
@@ -70,20 +77,27 @@ public function __construct($appName, IRequest $request, Manager $userManager, D
...
 		}
 		$loginResult = $this->userManager->checkPassword($user, $password);
 		if ($loginResult === false) {
-			$response = new Response();
+			$response = new JSONResponse();
 			$response->setStatus(Http::STATUS_UNAUTHORIZED);
 			return $response;
 		}
+
+		if ($this->twoFactorAuthManager->isTwoFactorAuthenticated($loginResult)) {
```
Comment (rullzer) [inline core/Controller/TokenController.php]: that naming is kind of weird...

Comment (ChristophWurst) [inline core/Controller/TokenController.php]: ``$loginResult``? :D

**Comment by ChristophWurst** [comment]
Updated. @rullzer @PVince81 please re-review ;-)

**Comment by PVince81** [comment]
Looks fine :+1:

**Comment by rullzer** [comment]
:+1: stands

**Pull request merged**